docs/usage/upstream-credentials.md

```diff
@@ -0,0 +1,68 @@
+# How to provide credentials for upstream repository
+
+In order to pull private images through registry cache, it is required to supply credentials for private upstream repository.
+
+## Configure registry cache to use upstream repository credentials
+
+### Create a `immutable` Secret with upstream repository credentials
+1. Encode credentials with base64 encoding:
+     ```bash
+     % echo "auser" | base64 -w 0
+     YXVzZXIK%
+     % echo "asecret" | base64 -w 0
+     YXNlY3JldAo=%
+     ```
+        > **Note** In some cases password can be a json (e.g. _json_key in **gcr** registry).
+        > If so, make sure the json value is enclosed in single quotes:
```

Comment:
Going over this once again -> to be honest, I think this is quite inconvenient. A Secret that you can use as imagePullSecret in a Pod, cannot be used as a registry cache Secret reference because the JSON is not enclosed in single quotes. 
I don't think it is the right thing to ask end users to take care of this. We should rather lift this limitation by proving the config as YAML and not as env vars. Let me try to experiment with this.

Reply:
Working WIP branch: https://github.com/ialidzhikov/gardener-extension-registry-cache/commits/config-file-approach

Reply:
We together authored 4e19e97b811d16ad319cb71eed25fe0962f889e5.